rosidl_generator_c/src/string_functions.c

```diff
@@ -48,10 +48,8 @@ rosidl_generator_c__String__fini(rosidl_generator_c__String * str)
         "Exiting.\n");
       exit(-1);
     }
-    if (str->data) {
```

Comment:
Because we're in an if statement that just checked if `str->data` was `NULL`, so there's no sense in checking it again.